### ISSUE TITLE
Unknown props `timeScale`, `timeFormat` on <g> tag

### ISSUE THREAD

**Issue by BrendanJones44** [esnet/react-timeseries-charts#171]
Hi, I'm working on implementing a time marker/tracker and I am getting this error when the chart loads:

Unknown props `timeScale`, `timeFormat` on <g> tag

I was following the code example for climate a bit: https://github.com/esnet/react-timeseries-charts/blob/master/src/website/packages/charts/examples/climate/Index.js

But here's my code:
`    renderMarker() {
        if(!this.state.tracker){
            return <g />;
        } else {
            return (
                <EventMarker
                    type="point"
                    axis="axis"
                    event={this.state.trackerEvent}
                    column="value"
                    info={[{ label: "Sprint Name", value: this.state.trackerStoryName}]}
                />
            );
        }
    }`

`        return(
            <Panel>
                <Resizable>
                    <ChartContainer timeRange={this.props.timeRange} width={1200} trackerPosition={this.state.tracker} onTrackerChanged={tracker => this.setState({tracker})}>
                        <ChartRow height="400">
                            <YAxis id="axis" label="STORY POINTS" min={0} max={yAxisMax} width="60" type="linear"/>
                            <Charts>
                                <LineChart axis="axis" style={lineStyle} series={this.props.series} interpolation="curveStepAfter"/>
                                {this.renderMarker()}
                            </Charts>
                        </ChartRow>
                    </ChartContainer>
                </Resizable>
            </Panel>
        );`

**Comment by pjm17971** [comment]
I think you can't just inject a `<g>` tag into the props (despite that example doing just that, except I think that case isn't hit). Anyway, it's trying to add props to it, assuming it is a user defined chart (it is supplying the `timeScale` etc from the `<ChartContainer>`, etc for that chart to render itself). But since the simple `<g>` tag doesn't expect those props, React is giving you that error.

Bottom line, if you change your code to just not include the marker rendering at all within <Charts> in that case, the problem should be solved.

**Comment by BrendanJones44** [comment]
Thank you very much for how responsive and active you've been. When doing your suggested change, I get the error: 
TypeError: Cannot read property 'props' of null

I believe it is because anything within the <Charts> tag gets parsed for props to display as a chart, but if there is no tracker, the {this.renderMarker()} returns null.

Is there another way you suggest handling this?

**Comment by pjm17971** [comment]
Yeah, that's kind of an issue I guess.

There's perhaps a couple of work arounds:
1) You can create a `Null` chart, perhaps just a simple functional component that returns a `<g />` as its render. `const NoTracker = () => <g />`. Then return that.
2) Or, you can restructure your code a bit so it you have one version of the whole chart with the marker and one without.

Right now the problem is you aren't adding nothing when `renderMarker()` returns null, you're adding a null as a child. That probably shouldn't break the code downstream, but it apparently doesn't detect that as a possibility and tries to use it as a chart.

Hope this helps.

**Comment by BrendanJones44** [comment]
Thank you! I got it working with just rendering null.
